community/security/src/main/java/org/neo4j/server/security/auth/UserSerialization.java

```diff
@@ -115,11 +129,18 @@ private Credential deserializeCredentials( String part, int lineNumber ) throws
 
     private String join( String separator, String... segments )
     {
+        return join(separator, Arrays.asList(segments).iterator() );
+    }
+
+    private String join( String separator, Iterator<String> segments )
```

Comment:
Please consider turning `segments` into stream and using joining collector instead of writing join logic once again.